pkg/engine/api/types.go

```diff
@@ -0,0 +1,49 @@
+package api
```

Comment:
This API is not final/stable/etc - naming needs discussion, to get these, I went with the types in each of the example files.
We'll probably want a single/exact file option along with the subdirectory